test/cpp/test_aten_xla_tensor.cpp

```diff
@@ -20,629 +16,622 @@ namespace cpp_test {
-  std::vector<at::Tensor> input_vars;
-  std::vector<at::Tensor> xinput_vars;
+    const std::vector<torch::Tensor>& inputs, const torch::Device& device,
+    const std::function<torch::Tensor(const std::vector<torch::Tensor>&)>& testfn,
+    double rtol = 1e-5, double atol = 1e-8) {
+  std::vector<torch::Tensor> input_vars;
+  std::vector<torch::Tensor> xinput_vars;
   for (size_t i = 0; i < inputs.size(); ++i) {
     auto& input = inputs[i];
     if (input.defined()) {
-      const bool requires_grad =
-          inputs_require_grad.empty() ? true : inputs_require_grad[i];
-      input_vars.push_back(
-          torch::autograd::make_variable(input, requires_grad));
-
-      at::Tensor xinput = bridge::CreateXlaTensor(CopyTensor(input), device);
-      xinput_vars.push_back(
-          torch::autograd::make_variable(xinput, requires_grad));
+      input_vars.push_back(input.clone().detach().set_requires_grad(input.requires_grad()));
+
+      torch::Tensor xinput = input.to(device, /*non_blocking=*/false, /*copy=*/true).detach().set_requires_grad(input.requires_grad());
+      xinput_vars.push_back(xinput);
     } else {
       input_vars.emplace_back();
       xinput_vars.emplace_back();
     }
   }
 
-  at::Tensor output = testfn(input_vars);
-  at::Tensor xoutput = testfn(xinput_vars);
+  torch::Tensor output = testfn(input_vars);
+  torch::Tensor xoutput = testfn(xinput_vars);
   output.backward();
   xoutput.backward();
-  for (size_t i = 0; i < input_vars.size(); ++i) {
-    if (inputs[i].defined() && input_vars[i].requires_grad()) {
+  for (size_t i = 0; i < inputs.size(); ++i) {
+    if (inputs[i].defined() && inputs[i].requires_grad()) {
       ASSERT_TRUE(xinput_vars[i].grad().defined());
       AllClose(input_vars[i].grad(), xinput_vars[i].grad(), rtol, atol);
     }
   }
 }
 
 TEST_F(AtenXlaTensorTest, TestScalarTensor) {
-  at::Tensor scalar_tensor =
-      at::scalar_tensor(1., at::TensorOptions(at::kFloat));
-  ForEachDevice([&](const Device& device) {
-    at::Tensor xla_scalar_tensor = at::scalar_tensor(
-        1., at::TensorOptions(at::kFloat).device(at::DeviceType::XLA));
+  torch::Tensor scalar_tensor =
+      torch::scalar_tensor(1., torch::TensorOptions(torch::kFloat));
+  ForEachTorchXLADevice([&](const torch::Device& device) {
+    torch::Tensor xla_scalar_tensor = torch::scalar_tensor(
+        1., torch::TensorOptions(torch::kFloat).device(torch::kXLA));
     AllClose(scalar_tensor, xla_scalar_tensor);
   });
 }
 
 TEST_F(AtenXlaTensorTest, TestClone) {
-  ForEachDevice([&](const Device& device) {
-    at::Tensor a = at::rand({2, 2}, at::TensorOptions(at::kFloat));
-    at::Tensor xla_a = bridge::CreateXlaTensor(a, device);
-    at::Tensor xla_b = xla_a.clone();
+  ForEachTorchXLADevice([&](const torch::Device& device) {
+    torch::Tensor a = torch::rand({2, 2}, torch::TensorOptions(torch::kFloat));
+    torch::Tensor xla_a = a.to(device, /*non_blocking=*/false, /*copy=*/true);
```

Comment:
This is quite verbose, and if there are changes to that API (and you guys do that :) ), we will have to refactor many lines of code.
I would have a wrapper instead.